log/event_print.go

```diff
@@ -44,15 +58,269 @@ func getHeaderSubsection(key, value string) string {
 func getHeaderLine(content string) string {
 	// The available space is 4 char less because if the `| ` prefix and ` |` suffix every line has.
 	contentMaxWidth := stepRunSummaryBoxWidthInChars - 4
-	widthDiff := len(content) - contentMaxWidth
+	return widthConstrainedStringWithBorder(content, contentMaxWidth)
+}
+
+func widthConstrainedStringWithBorder(content string, width int) string {
+	return fmt.Sprintf("| %s |", widthConstrainedString(content, width))
+}
+
+func widthConstrainedString(content string, width int) string {
+	widthDiff := len(content) - width
 
 	if widthDiff < 0 {
-		return fmt.Sprintf("| %s%s |", content, strings.Repeat(" ", -widthDiff))
+		return fmt.Sprintf("%s%s", content, strings.Repeat(" ", -widthDiff))
 	}
 
 	if widthDiff == 0 {
-		return fmt.Sprintf("| %s |", content)
+		return content
+	}
+
+	return stringutil.MaxFirstCharsWithDots(content, width)
+}
+
+// This is the main entry point to generate the finished event console log lines.
+func generateStepFinishedFooterLines(params StepFinishedParams) []string {
+	mainSeparator := fmt.Sprintf("+%s+%s+%s+", strings.Repeat("-", footerIconBoxWidth), strings.Repeat("-", footerTitleBoxWidth), strings.Repeat("-", footerExecutionTimeBoxWidth))
+	sectionSeparator := fmt.Sprintf("|%s|", strings.Repeat(" ", stepRunSummaryBoxWidthInChars-2))
+
+	deprecated := false
+	if params.Deprecation != nil {
+		deprecated = params.Deprecation.RemovalDate != "" || params.Deprecation.Note != ""
+	}
+
+	var lines []string
+	lines = append(lines, sectionSeparator)
+	lines = append(lines, mainSeparator)
+	lines = append(lines, getSummaryFooterRow(params.InternalStatus, params.Title, params.StatusReason, params.RunTime, deprecated))
+	lines = append(lines, mainSeparator)
+
+	hasPreviousSection := false
+
+	if len(params.Errors) > 0 {
+		hasPreviousSection = true
+		lines = append(lines, getIssueAndSourceSection(params.SupportURL, params.SourceCodeURL)...)
+	}
+
+	if params.Update != nil {
+		updateAvailable, _ := utils.IsUpdateAvailable(params.Update.ResolvedVersion, params.Update.LatestVersion)
+		if updateAvailable {
```

Comment:
do we need this check? it seem the same check was done when params.Update got a value: https://github.com/bitrise-io/bitrise/pull/826/files#diff-5844392ecf207d0af644ed8fa1fb13c10326a3014f83ac0325ec858fc03e211dR186

Reply:
You are right. The nature of this update struct is that if it exists then there is an update available otherwise there is not. This is an unnecessary check here and I will remove it.